airflow/jobs/job.py

```diff
@@ -126,7 +127,19 @@ def executor(self):
     def heartrate(self) -> float:
         return Job._heartrate(self.job_type)
 
-    def is_alive(self, grace_multiplier=2.1) -> bool:
+    @cached_property
```

Comment:
Will we be able to add a unit test for this?

Reply:
let me check this one @amoghrajesh

Reply:
@amoghrajesh this function is already covered in the unit test. the right function for adding the unit test would be heartbeat(). Let me check that

Reply:
@amoghrajesh i have added the unit test for heartbeat function.